internal/cmd/controller/gitops/reconciler/status_controller.go

```diff
@@ -110,6 +113,28 @@ func (r *StatusReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		gitrepo.Status.Display.State = "GitUpdating"
 	}
 
+	// We're explicitly setting the ready status from a bundle here, but only if it isn't ready.
+	//
+	// - If the bundle has no deployments, there is no status to be copied from the setStatus
+	// function, so that we won't overwrite anything.
+	//
+	// - If the bundle has rendering issues and there are deployments of which there is at least one
+	// in a failed state, the status of the bundle deployments would be overwritten by the bundle
+	// status.
+	//
+	// - If the bundle has no rendering issues and there are deployments in a failed state, the code
+	// will overwrite the gitreop status condition with the same status from the bundle. This is why
```

Comment:
```suggestion
	// will overwrite the gitrepo status condition with the same status from the bundle. This is why
```

I am not entirely sure of understanding this sentence, especially the `same status from the bundle` bit: this means to say that the bundle's status will be propagated to the gitrepo, right?

Reply:
I did adapt the text though, hoping that it might be clearer now.

Reply:
That's clearer, thanks!